src/fifo.rs

```diff
@@ -0,0 +1,197 @@
+        let mut payload: u8 = ctrl_reg9 & !FIFO_Bitmasks::CTRL_REG9_FIFO;
+        payload |= data;
+        self.interface.write(
+            Sensor::Accelerometer,
+            register::AG::CTRL_REG9.addr(),
+            payload,
+        )?;
+
+        Ok(())
+    }
+
+    /// Get flags and FIFO level from the FIFO_STATUS register
+    pub fn get_fifo_status(&mut self) -> Result<FifoStatus, T::Error> {
+        let fifo_src = self.read_register(Sensor::Accelerometer, register::AG::FIFO_SRC.addr())?;
+        let fifo_level_value = fifo_src & FIFO_Bitmasks::FSS;
+        let status = FifoStatus {
+            /// Is FIFO filling equal or higher than the threshold?
+            fifo_thresh_reached: match fifo_src & FIFO_Bitmasks::FTH {
+                0 => false,
+                _ => true,
+            },
+            /// Is FIFO full and at least one sample has been overwritten?
+            fifo_overrun: match fifo_src & FIFO_Bitmasks::OVRN {
+                0 => false,
+                _ => true,
+            },
+            /// Is FIFO empty (no unread samples)?
+            fifo_empty: match fifo_level_value {
+                0 => true,
+                _ => false,
+            },
+            /// Read FIFO stored data level
+            fifo_level: fifo_level_value,
+        };
+        Ok(status)
+    }
+
+    pub fn set_decimation(&mut self, decimation: DECIMATE) -> Result<(), T::Error> {
+        let data: u8 =
+            self.read_register(Sensor::Accelerometer, register::AG::CTRL_REG5_XL.addr())?; // read current content of the register
+        let mut payload: u8 = data & !FIFO_Bitmasks::DEC; // use bitmask to affect only bits [7:6]
+        payload |= decimation.value(); // set the selected decimation value
+        self.interface.write(
+            Sensor::Accelerometer,
+            register::AG::CTRL_REG5_XL.addr(),
+            payload,
+        )?;
+        Ok(())
+    }
+}
+
+/// FIFO mode selection. (Refer to datasheets)
+#[allow(non_camel_case_types)]
+#[derive(Debug, Clone, Copy)]
+pub enum FIFO_MODE {
+    /// Bypass mode (he FIFO is not operational and it remains empty).
+    Bypass = 0b000,
+    /// FIFO mode (data from the output channels are stored in the FIFO until it is overwritten).
+    FIFO = 0b001,
+    /// Continuous-to-FIFO mode (continuous mode until trigger is deasserted, then FIFO mode).
+    Continuous_to_FIFO = 0b011,
+    /// Bypass-to-Continuous mode (Bypass mode until trigger is deasserted, then Continuous mode).
+    Bypass_to_continuous = 0b100,
+    /// Continuous mode. If the FIFO is full, the new sample overwrites the older sample.
+    Continuous = 0b110,
+}
+
+impl FIFO_MODE {
+    pub fn value(self) -> u8 {
+        (self as u8) << 5 // shifted into the right position, can be used directly
+    }
+}
+
+/// Decimation of acceleration data on OUT REG and FIFO (Refer to table 65)
+#[allow(non_camel_case_types)]
+#[derive(Debug, Clone, Copy)]
+pub enum DECIMATE {
```

Comment:
Does `DECIMATE` have to be all uppercase? Can it be `Decimate`?

Reply:
Good question. I see now that I wasn't consistent about it and some enums are ALL CAPS, while others are not. I tried checking here: [naming conventions](https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md), but it only mentions enum variants specifically. But I guess enum is closer to types and traits, so it probably should be UpperCamelCase. If we agree on that, there's some more to change accross the code.